pmpro-sponsored-members.php

```diff
@@ -1722,9 +1722,52 @@ function pmprosm_profile_fields_seats($user)
 add_action('edit_user_profile', 'pmprosm_profile_fields_seats');
 
 function pmprosm_display_sponsored_accounts($member_ids) {
-
-    //make sure we have something to display
-	if ( empty( $member_ids) ) return '';
+        $request = $_REQUEST;
+        /*
+         * Get the current logged in user for the front end
+         * or
+         * Get user id for editing the user-edit.php page backend
+         */
+        $user_id = get_current_user_id();
+        if(isset($request['user_id'])){
+            $user_id = $request['user_id'];
+        }
+        
+        /*
+         * For removing members from seats
+         */
+        if(isset($request['member_id']) && isset($request['levelstocancel']) && isset($request['action'])){
+            /*
+             * check nonce for security
+             */
+            if(wp_verify_nonce($request['action'], 'remove-member')){
+                /*
+                 * Removes member from seat
+                 */
+                $reponse = pmprosm_remove_member_from_seat($request['member_id'], $request['levelstocancel'], $user_id);

```

Comment:
use intval to sanitize these.

I usually set all the variables I'm going to use in one section near the top of the method.